api/src/main/java/org/openmrs/module/auditlog/service/impl/AuditLogServiceImpl.java

```diff
@@ -30,11 +32,13 @@ public AuditLogServiceImpl(AuditLogDao auditLogDao) {
         this.auditLogDao = auditLogDao;
     }
 
+    @Transactional(readOnly=true)
     public ArrayList<SimpleObject> getLogs(String username, String patientId, Date startDateTime, Integer lastAuditLogId, Boolean prev, Boolean defaultView) {
         List<AuditLog> auditLogs = auditLogDao.getLogs(username, patientId, startDateTime, lastAuditLogId, prev, defaultView);
         return (ArrayList<SimpleObject>) (auditLogs.stream().map(AuditLog::map).collect(Collectors.toList()));
     }
 
+    @Transactional
```

Comment:
Hi @shilpa-iplit,
do you need `@Transactional` for each method in this class (as you already define `@Transactional` for  this class)?